IllegalArgumentException: No enclosing component class found for interface

**Stacktrace:**
```
Caused by: java.lang.IllegalArgumentException: No enclosing component class found for interface com.example.ExampleComponent$Builder
  at com.yandex.yatagan.common.Loader__LoadImplementationKt.loadImplementationByBuilderClass(SourceFile:31)
  at com.yandex.yatagan.common.Loader.loadImplementationByBuilderClass(SourceFile:1)
  at com.yandex.yatagan.Yatagan.builder(SourceFile:51)
```

**Usage:**
Component:
```
package com.example

@Component
interface ExampleComponent {
    
   @Component.Builder
    interface Builder {
        fun build()
    }
}
```
Component usage:
```
val component = Yatagan.builder(ExampleComponent::class.java).build()
```

**Environment:**
Android Gradle Plugin 8.1.0
R8 8.1.59
openjdk 17.0.8.1

--- a/api/common/src/main/kotlin/loadImplementation.kt
+++ b/api/common/src/main/kotlin/loadImplementation.kt
@@ -28,9 +28,11 @@ fun <T : Any> loadImplementationByBuilderClass(builderClass: Class<T>): T {
     require(builderClass.isAnnotationPresent(Component.Builder::class.java)) {
         "$builderClass is not a builder for a Yatagan component"
     }
-    val componentClass = requireNotNull(builderClass.enclosingClass) {
+    val componentClassName = builderClass.name.substringBeforeLast("$")
+    require(componentClassName != builderClass.name) {
         "No enclosing component class found for $builderClass"
     }
+    val componentClass = builderClass.classLoader.loadClass(componentClassName)
     val yataganComponentClass = loadImplementationClass(componentClass)
     return builderClass.cast(yataganComponentClass.getDeclaredMethod("builder").invoke(null))
 }

